DEHPEcosimPro/ViewModel/Dialogs/HubMappingConfigurationDialogViewModel.cs

```diff
@@ -438,21 +576,213 @@ private void UpdateProperties()
+            }
+
+            var arrayVariables = availableVariables.Where(x => x.Reference.DisplayName.Text.Contains("[")).ToList();
+
+            this.AvailableVariables.AddRange(availableVariables.Where(x => arrayVariables
+                    .All(y => y.Reference.NodeId.Identifier != x.Reference.NodeId.Identifier))
+                .Select(x =>
+                {
+                    var (reference, _) = x;
+                    return new VariableRowViewModel((reference, this.DstController.ReadNode(x.Reference)));
+                }));
+
+            var grouby = arrayVariables.GroupBy(x => this.GetNameFromArrayReference(x.Reference));
+
+            foreach (var arrayVariablesGrouped in grouby)
+            {
+                if (arrayVariablesGrouped.Key == null)
+                {
+                    continue;
+                }
+
+                if (arrayVariablesGrouped.ToList().Count < 2)
+                {
+                    this.AvailableVariables.AddRange(arrayVariablesGrouped
+                        .Select(x =>
+                        {
+                            var (reference, _) = x;
+                            return new VariableRowViewModel((reference, this.DstController.ReadNode(x.Reference)));
+                        }));
+                } 
+                else
+                { 
+                    this.AvailableVariables.Add(new ArrayVariableRowViewModel(arrayVariablesGrouped.Key, arrayVariablesGrouped
+                    .Select(x => new VariableRowViewModel((x.Reference, this.DstController.ReadNode(x.Reference))))));
+
+                }
+            }
+
+            this.MappedElements.ChangeTrackingEnabled = true;
+
+            this.IsBusy = false;
+        }
+
+        /// <summary>
+        /// </summary>
+        public void SelectedAvailableVariablesChanged()
+        {
+            this.AreVariableTypesAreCompatible();
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="reference"></param>
+        /// <returns></returns>
+        private string GetNameFromArrayReference(ReferenceDescription reference)
+        {
+            var splitedName = reference.BrowseName.Name.Split('[', ']');
+
+            if (splitedName.Length >= 1 && splitedName.Length <= 2)
+            {
+                return splitedName[0];
+            }
+
+            if (splitedName.Length > 2)
+            {
+                return splitedName[0] + splitedName[2];
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="variable"></param>
+        /// <param name="parameter"></param>
+        /// <param name="compatibilityOfColumns"></param>
+        /// <returns></returns>
+        private bool AreArraysCompatible(VariableBaseRowViewModel variable, ParameterOrOverrideBase parameter)
+        {
+            var rowsAndColumnsOfVariable = variable.ActualValue.ToString().Split('[', 'x', ']');
```

Comment:
This is not the right way to determine sizes. You have to look at the actual underlying model elements.